jib-core/src/main/java/com/google/cloud/tools/jib/frontend/CredentialRetrieverFactory.java

```diff
@@ -143,33 +149,25 @@ public CredentialRetriever dockerCredentialHelper(Path credentialHelper) {
   }
 
   /**
-   * Creates a new {@link CredentialRetriever} that tries common Docker credential helpers to
+   * Creates a new {@link CredentialRetriever} that tries well-known Docker credential helpers to
    * retrieve credentials based on the registry of the image, such as {@code docker-credential-gcr}
-   * for images with the registry as {@code gcr.io}.
+   * for images with the registry ending with {@code gcr.io}.
    *
    * @return a new {@link CredentialRetriever}
    */
-  public CredentialRetriever inferCredentialHelper() {
-    List<String> inferredCredentialHelperSuffixes = new ArrayList<>();
-    for (String registrySuffix : COMMON_CREDENTIAL_HELPERS.keySet()) {
-      if (!imageReference.getRegistry().endsWith(registrySuffix)) {
-        continue;
-      }
-      String inferredCredentialHelperSuffix = COMMON_CREDENTIAL_HELPERS.get(registrySuffix);
-      if (inferredCredentialHelperSuffix == null) {
-        throw new IllegalStateException("No COMMON_CREDENTIAL_HELPERS should be null");
-      }
-      inferredCredentialHelperSuffixes.add(inferredCredentialHelperSuffix);
-    }
+  public CredentialRetriever wellKnownCredentialHelpers() {
```

Comment:
Sorry, I retract my previous comment :-(. This should be singular, since we return a single element.  Maybe `findWellKnownCredentialHelper()`?

Reply:
I actually think the plural makes sense. It is not returning cred helper(s) but rather a retriever, which may attempt more than one cred helper. For example, the mock instance in the test for this is called `mockWellKnownCredentialHelpersCredentialRetriever`.

And using a verb makes sense, but because all the methods of this class don't use verb (`dockerConfig()`, `known()`, `dockerCredentialHelper()`, and `googleApplicationDefaultCredentials()`), I will retain without-verb names. Maybe a separate PR.

Reply:
ok